lib/mods/date.rb

```diff
@@ -181,16 +189,77 @@ def self.cleanup(text)
       end
     end
 
+    # Full-text extractor for data formatted as YYYY-YYYY
+    class YearRangeFormat < ExtractorDateFormat
+      REGEX = /(?<start>\d{3,4})-(?<end>\d{3,4})/
+
+      def self.cleanup(text)
+        matches = text.match(REGEX)
+        "#{matches[:start].rjust(4, "0")}/#{matches[:end].rjust(4, "0")}"
+      end
+    end
+
+    # Full-text extractor for data formatted as YYY-
+    class DecadeAsYearDashFormat < ExtractorDateFormat
+      REGEX = /(?<year>\d{3})[-_x?](?!\d)/
+
+      def self.cleanup(text)
+        matches = text.match(REGEX)
+        "#{matches[:year]}X"
+      end
+    end
+
+    # Full-text extractor that tries hard to pick any year present in the data
+    class EmbeddedBCYearFormat < ExtractorDateFormat
+      REGEX = /(?<year>\d{3,4})\s?B\.?C\.?/i
+
+      def self.cleanup(text)
+        matches = text.match(REGEX)
+        "-#{(matches[:year].to_i - 1).to_s.rjust(4, "0")}"
+      end
+    end
+
     # Full-text extractor that tries hard to pick any year present in the data
     class EmbeddedYearFormat < ExtractorDateFormat
-      REGEX = /(?<prefix>-)?(?<year>\d{3,4})/
+      REGEX = /(?<prefix>-)?(?<!\d)(?<year>\d{4})(?!\d)/
```

Comment:
Can you give an example of what a prefix might be?

Reply:
A literal `-`?

Reply:
👍